### DISCUSSION TITLE
Default ESLint config with typescript should include typescript-eslint:recommended(-requiring-type-checking)

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [remix-run/remix#6126] [Proposals]
Similar to #5071, but for projects that enable TypeScript. Right now, Remix projects created with `npx create-remix@latest` and TypeScript use the TSESLint parser for TypeScript projects -great!- but haven't enabled any of its [preset configurations](https://typescript-eslint.io/docs/linting/configs). Those configurations are recommended for most-to-all TypeScript projects by the TSESLint maintenance team.

To verify the list of ESLint rules used in an app newly created with `create-remix --typescript`, you can run:

```sh
npx eslint --print-config app/routes/_index.tsx > rules.json
```

Is there a particular reason not to include the [`@typescript-eslint/recommended`](https://typescript-eslint.io/linting/configs#recommended) and [`@typescript-eslint/recommended-requiring-type-checking`](https://typescript-eslint.io/linting/configs#recommended-requiring-type-checking) configs for users by default? 

A couple of notes on why that's likely more desirable now than in the past:

* The recommended configs should no longer contain opinionated rules that wouldn't be good in most projects
* If there are still overly chafe-y rules, we're gearing up for a 6.X release and would be happy to move those to the new [strict configuration](https://typescript-eslint.io/docs/linting/configs#strict)

~Proposed solution:~

1. ~Make a `@remix-run/typescript` lint config in [`@remix-run/eslint-config`](https://www.npmjs.com/package/@remix-run/eslint-config) that extends from `@typescript-eslint/recommended` and `@typescript-eslint/recommended-requiring-type-checking`~
2. ~In `create-remix`, when TypeScript is provided, change the `.eslintrc.json` file contents to include `"@remix-run/typescript"` in `extends`.~

Edit based on https://github.com/remix-run/remix/discussions/6126#discussioncomment-5758806: there's already a `typescript` set of overrides! So this discussion should be around enabling recommended rules in them.

**Comment by MichaelDeBoey** [comment]
As you can see in the default config, we're already including TypeScript rules

https://github.com/remix-run/remix/blob/66b39d1f8538b8569f6d1cfa08a2c0c3d5a3e754/packages/remix-eslint-config/index.js#L77
https://github.com/remix-run/remix/blob/66b39d1f8538b8569f6d1cfa08a2c0c3d5a3e754/packages/remix-eslint-config/rules/typescript.js#L5-L54

So I don't think we need an extra `typescript` entry point tbh.

If you want to extend the configs you mentioned by default, I would suggest to create a PR that does so and then the team will look into if it's desired or not.

**Reply by JoshuaKGoldberg** [reply]
> So I don't think we need an extra `typescript` entry point tbh.

Aha excellent! Apologies, I'd missed this. I'll edit the OP to make it more clear that the right proposal is augmenting the existing config, not crafting a new config.

> already including TypeScript rules

It's including _some_ TypeScript rules, yes. But there are dozens of other rules that come with each of [`plugin:@typescript-eslint/recommended`](https://typescript-eslint.io/linting/configs#recommended) and the other configs. The Remix Typescript config right now has fewer than a dozen enabled. It's missing out!

_(apologies if you already know this - just posting for visibility, as this has been a frequent area of confusion for users)_

**Comment by brophdawg11** [comment]
https://github.com/remix-run/remix/pull/6614 is merged so this should be resolve din the next release (`1.18.0`)

**Reply by JoshuaKGoldberg** [reply]
Amazing, thanks @brophdawg11! I really appreciate you & co. working with me on this. And I'll send a PR with typescript-eslint v6 improvements, too, once we get it out the door!
